### PR TITLE
Add parameter to provide internal region specific cache scrape interval

### PR DESCRIPTION
**Pull request by mithunmgore**

### CODE REVIEW

**Comment by mithunmgore** [comment]
@viennaa @christopherhans 
want to have different internal cache scrape interval for each region. Problem is, currently the ucs-exporter getting restart in the regions where ucsm servers are more than approx 10 servers (eu-de-1, eu-de-2).

File: prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ spec:
              - {{ required ".Values.ucs.user missing" .Values.ucs.user }}
              - -p
              - {{ required ".Values.global.exporter_master_password missing" .Values.global.exporter_master_password }}
+             - -s
+             - {{ .Values.ucs.internal_cache_scrape_interval }}
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:32]: Please add this mapping in `values.yaml` with a default value for "normal" regions.

**Review by viennaa — CHANGES_REQUESTED** [review]

File: prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml (hunk 1 above)
Comment (BerndKue) [reply prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:32]: What happens if not defined in secrets?
How about writing therefor
{{ if .Values.ucs.internal_cache_scrape_interval }}{{ .Values.ucs.internal_cache_scrape_interval }}{{ else }}300{{ end }}

Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:32]: This one is not specified as "required", so exporter will go with default value(300), I don't think loop is required to specify this. 
@viennaa please confirm.

Comment (viennaa) [reply prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:32]: The whole setting starting with `-s` need to be in the if else block. Default is 300, so nothing need to be specified indeed: https://github.com/sapcc/ucs-exporter/blob/master/ucs-exporter.py#L51-L52

So I'd go with 

```
{{ if .Values.ucs.internal_cache_scrape_interval }}
- -s
- {{ .Values.ucs.internal_cache_scrape_interval }}
{{ end }}
```

See comment in values.yaml.

File: prometheus-exporters/ucs-exporter/values.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ ucs:
    query: "ucsa"
    status: "active"
    debug: "0"
+   internal_cache_scrape_interval: DEFINED-IN-REGION-SECRETS
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/values.yaml:19]:
```suggestion
   # defaults to 300, only use if needed
   # internal_cache_scrape_interval: DEFINED-IN-REGION-SECRETS 
```

**Comment by BerndKue** [comment]
@mithunmgore If you make the changes like proposed by Tommy, I would be fine with merging.

**Comment by mithunmgore** [comment]
Done.

**Review by viennaa — APPROVED** [review]

**Comment by BerndKue** [comment]
@mithunmgore Thanks a lot!

**Pull request merged**